webapp/src/components/properties/link/link.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode} from 'react'
+
+import Editable from '../../../widgets/editable'
+
+import './link.scss'
+import {Utils} from '../../../utils'
+
+type Props = {
+    value: string
+    onChange: (value: string) => void
+    onSave: () => void
+    onCancel: () => void
+    validator: (newValue: string) => boolean
+}
+
+const URLProperty = (props: Props): JSX.Element => {
+    let link: ReactNode = null
+    if (props.value?.trim()) {
+        link = (
+            <a
+                className='icon-link-variant'
```

Comment:
This introduces another way to show icons, we were using icon components, and here you are using a class to introduce the icons, in my opinion this is more confusing and hard to track, I prefer to avoid that and create a new icon component that just render a `<span className='icon-link-variant'/>`, that way is easier to update/customize icons in a homogeneous way, and also we have a clear place to find all the supported icons in the application.

Reply:
@jespino The existing icon components embedded an SVG, which happened before we added compass icons to Focalboard. With Compass Icons, I think this is the simplest way. If we create a layer of React components over compass icons, we'd have to keep creating new components each time we use. anew icons. Also, with the plan to make compass icons an NPM package and simple add it as a dependency, these components will need maintain wither we update the compass icons.

This is also how Mattermost uses compass icons. React component would make sense if for some reason we wanted to encapsulate the implementation of icons. Otherwise it's unnecessary components IMO.

What's you're rational for using components for individual icons BTW? Maybe I'm missing something?

Reply:
Added React components for individual icons as discussed.